src/main/scala/is/hail/HailContext.scala

```diff
@@ -31,6 +31,10 @@ case class FilePartition(index: Int, file: String) extends Partition
 
 object HailContext {
 
+  private var cachedContext: HailContext = _
```

Comment:
maybe theContext, or just context?  You're not caching it, you're just storing it.